$MirOS$

	• only define MAX if it isn’t already defined
	• revert an upstream patch, as it causes ICEs

--- mip/regs.c.orig	Thu Jan  1 00:00:00 1970
+++ mip/regs.c	Fri May  2 17:56:59 2008
@@ -41,7 +41,9 @@
 
 #define	MAXLOOP	20 /* Max number of allocation loops XXX 3 should be enough */
 
+#ifndef MAX
 #define MAX(a,b) (((a) > (b)) ? (a) : (b))
+#endif
  
 /*
  * New-style register allocator using graph coloring.
@@ -2034,7 +2036,6 @@ static REGW *spole;
 static void
 longtemp(NODE *p)
 {
-	NODE *l, *r;
 	REGW *w;
 
 	if (p->n_op != TEMP)
@@ -2047,10 +2048,9 @@ longtemp(NODE *p)
 			w->r_color = BITOOR(freetemp(szty(p->n_type)));
 			w->r_class = 1;
 		}
-		l = mklnode(REG, 0, FPREG, INCREF(p->n_type));
-		r = mklnode(ICON, w->r_color, 0, INT);
-		p->n_left = mkbinode(PLUS, l, r, INCREF(p->n_type));
-		p->n_op = UMUL;
+		p->n_op = OREG;
+		p->n_lval = w->r_color;
+		p->n_rval = FPREG;
 		p->n_regw = NULL;
 		break;
 	}
